### PR TITLE
Migration error bookmarks

### PR DESCRIPTION
**Pull request by vauxhall**
Made a test but had to comment it as I noticed it is flaky and sometimes the `RustPlaces` can't load the bookmarks; I opted to leave the code commented instead of delete as it can be improved as I learn more about RustPlaces.

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -5920,20 +5922,22 @@
 		F84B21D61A090F8100AAB793 /* ClientTests */ = {
 			isa = PBXGroup;
 			children = (
-				43446CF12412F9A800F5C643 /* ETPCoverSheet */,
```
Comment (ecotopian) [inline Client.xcodeproj/project.pbxproj:5923]: Did you sort this folder in XCode? Usually that's not a bad idea but I suggest to revert this to not run into future merge conflicts of the .pbxproj file

File: Client/Ecosia/Migration/EcosiaFavourites.swift [hunk 2]
```diff
@@ -59,3 +61,10 @@ final class EcosiaFavourites {
     }
 
 }
+
+private extension Core.Page {
+    var urlString: String? {
```
Comment (ecotopian) [inline Client/Ecosia/Migration/EcosiaFavourites.swift:66]: what about adding a unit-test for this accessor?

File: ClientTests/EcosiaBookmarkTests.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+//                            }
+//                            
+//                            _ = profile.places.forceClose()
+//                            try? profile.files.remove("profile-test_places.db")
+//                        }
+//                    }
+//                }
+//            }
+//        }
+//        waitForExpectations(timeout: 20)
```
Comment (ecotopian) [inline ClientTests/EcosiaBookmarkTests.swift:46]: yeah testing this is tricky.. what you could do is leave the code uncommented but just deactivate this Test in the scheme. That way we can still it run it locally in XCode without the need to change the code :)

Comment (ecotopian) [reply ClientTests/EcosiaBookmarkTests.swift:46]: ..and it's not run on CI then

File: Client.xcodeproj/project.pbxproj (hunk 1 above)
Comment (vauxhall) [reply Client.xcodeproj/project.pbxproj:5923]: ok

File: Client/Ecosia/Migration/EcosiaFavourites.swift (hunk 2 above)
Comment (vauxhall) [reply Client/Ecosia/Migration/EcosiaFavourites.swift:66]: Sure

File: ClientTests/EcosiaBookmarkTests.swift (hunk 3 above)
Comment (vauxhall) [reply ClientTests/EcosiaBookmarkTests.swift:46]: That's a good idea

File: ClientTests/EcosiaBookmarkTests.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                            try? profile.files.remove("profile-test_places.db")
+                        }
+                    }
+                }
+            }
+        }
+        waitForExpectations(timeout: 20)
+    }
+    
+    func testURLString() {
```
Comment (ecotopian) [inline ClientTests/EcosiaBookmarkTests.swift:50]: 👍

**Review by ecotopian — APPROVED** [review]
Awesome! 🥇

**Pull request merged**